content/reference/wadm/api.md

```diff
@@ -0,0 +1,254 @@
+
+## Topic Space
+The wadm API is exposed entirely as a NATS service on a topic space. All of the API operations will occur as _requests_ on a topic in the following format:
+
+```
+wadm.api.{lattice-id}.{category}.{operation}.{object}
+```
+
+The `operation` is usually a verb, and `object` is an optional scope-limiter to the operation. In many cases, the `object` will be something like a model name.
+
+All requests and responses on this topic are encoded as JSON.
+
+If a consumer wishes to monitor events in real-time (for example, you want to be notified as soon as compensating actions are taken), then it can subscribe to the following topic:
+
+```
+wadm.evt.{lattice-id}
+```
+
+The events on this topic are JSON-encoded [CloudEvents](https://cloudevents.io/) (current spec version 1.0.1 as of May 1, 2022).
+
+**NOTE** that all model _names_ are treated like unique identifiers and must conform to the rules governing NATS topic segments. For example, they cannot contain spaces, commas, unprintable characters, or periods.
+
+## Model Persistence
+The following operations pertain to storing and retrieving models. Persistence of models is explicitly and deliberately separated from deployment management.
+
+
+### Get Model List
+`wadm.api.{lattice}.model.list`
+
+Retrieves a list of models within the given lattice. The status of the model is an aggregate of the statuses of the traits defined within. Only the newest version of the model is included in this list.
+
+**Request**: `none`
+
+**Response**: 
+```json
+{
+    "name": "..",
+    "version": "..",
+    "description": "..",
+    "deployed": true,
+    "status": "compensating|ready|failed|undeployed"
+}
+```
+
+### Get a Model Spec
+`wadm.api.{lattice}.model.get.{name}`
+
+Retrieves the specification of the model stored with the given name. Caller must
+specify the version of the model to be displayed. A list of stored versions can be obtained with the `versions` verb.
+
+**Request**:
+```json
+{
+    "version": "1.0"
+}
+```
+
+**Response**:
+
+JSON serialization of [OAM model/yaml](https://github.com/wasmCloud/wadm/tree/main/oam)
+
+
+
+### Store Models
+`wadm.api.{lattice}.model.put.{name}`
+
+Model storage is _append-only_. New versions are added to the model's version history according to retention policy and will not replace previously existing versions. If the model and version being submitted already exist, the request will be _rejected_. Note that this won't automatically deploy a model, it only affects storage. The response will tell the caller how many versions are on file and the current version number _after_ the operation completed.
+
+**Request**: JSON serialization of OAM model
+
+**Response**:
+```json
+{
+    "result": "error|created|newversion",
+    "total_versions": 4,
+    "current_version": "1.0",
+    "message": "..."
+}
+```
+
+The response will indicate if storage failed, if a new model was created, or if a new version of a model was pushed.
+
```

Comment:
Are there any requirements on the syntax of the version field provided by the caller? Is latest version determined by timestamp of submission or by the highest version number?

Reply:
I suppose  syntax is dictated by OAM `metadata.annotations.version` - if so nevermind on the first part of the question. Still would like to know about whether the field or timestamp are used to determine 'latest'